server/src/domain/metadata/metadata.service.ts

```diff
@@ -340,6 +340,31 @@ export class MetadataService {
     const stats = await this.storageRepository.stat(asset.originalPath);
     const mediaTags = await this.repository.getExifTags(asset.originalPath);
     const sidecarTags = asset.sidecarPath ? await this.repository.getExifTags(asset.sidecarPath) : null;
+
+    const dateDateSidecarTags = firstDateTime(sidecarTags as Tags, [
+      'SubSecDateTimeOriginal',
+      'DateTimeOriginal',
+      'SubSecCreateDate',
+      'CreationDate',
+      'CreateDate',
+      'SubSecMediaCreateDate',
+      'MediaCreateDate',
+      'DateTimeCreated',
+    ]);
+
+    if (dateDateSidecarTags && mediaTags) {
+      // remove time from mediaTags, since sidecarTags has a more accurate time
+
+      mediaTags.SubSecDateTimeOriginal = undefined;
+      mediaTags.DateTimeOriginal = undefined;
+      mediaTags.SubSecCreateDate = undefined;
+      mediaTags.CreationDate = undefined;
+      mediaTags.CreateDate = undefined;
+      mediaTags.SubSecMediaCreateDate = undefined;
+      mediaTags.MediaCreateDate = undefined;
+      mediaTags.DateTimeCreated = undefined;
+    }
+
```

Comment:
This isn't strictly related to writing exif, but the idea that a sidecar file should be able to overwrite date/time fields from the sidecar. This is a better implementation of #4502. If possible, can you separate this change out into a separate PR, which is probably easier to merge, sooner.

Reply:
Didn't know it was a know issue, a new PR is open on #5199

Reply:
Should I remove it from here ?

Reply:
Yeah, you can remove and then we'll merge that other one pretty quick and you can rebase this PR to get those changes.